007-OFEP-flag-change-events.md

```diff
@@ -0,0 +1,74 @@
+## 007-OFEP-flag-change-events
+
+## State: DRAFTING
+
+Some flag SDKs support listening for flag value changes or general configuration changes ([launchdarkly](https://docs.launchdarkly.com/sdk/features/flag-changes), [cloudbees](https://docs.cloudbees.com/docs/cloudbees-feature-management/latest/reporting/configuration-fetched-handler), flagd). This can allow us to use an event-based paradigm for consuming flags. Client apps may use feature flags for characteristics that aren't specifically tied to a user-action, making imperative flag evaluation a less-than-ideal solution. A server application could listen to a flag that changes some operational behavior.
+
+Examples:
+
+- A web client's banner color could be updated when an associated flag is changed, without any user action or a page reload.
+- A service could subscribe to a flag that controls its global log-level.
+- A subsystem could be restarted with the new flag value.
+
+## Design
+
+The provider interface and the OpenFeature client would be extended to have new functionality to register handlers for a set of events defined by the SDK (`ProviderEvents`). When a _application author_ registers a handler on a client, the client the client maintains this handler. The provider emits events or runs a callback indicating that it received a certain event, optionally providing data associated with that event. The callbacks registered with the client are then invoked with this data (`EventData`).
```

Comment:
Good point. I've added `application integrator`. Can you help me understand how an [integration author](https://docs.openfeature.dev/docs/specification/glossary#integration-author) (probably somebody who writes hooks) might use this API?

Reply:
The example that was top of mind for me was someone who's creating a react-specific feature flagging API that's built on top of Open Feature. I think that fits the persona of Integration Author? This is top of mind because I've been creating [exactly that API](https://github.com/moredip/openfeature-playground/blob/e41e472f45eff9b36a54ceb5ff66e8e1546c77c2/libs/openfeature-hooks/src/lib/useFlag.ts#L24) (a react hook for Open Feature), and need these events in order to make it work in an idiomatic react way.

Reply:
You're absolutely right. I was being myopic thinking of hooks. Will add that persona.